lib/iris/_data_manager.py

```diff
@@ -76,12 +79,69 @@ def __deepcopy__(self, memo):
         """
         return self._deepcopy(memo)
 
+    def __eq__(self, other):
+        """
+        Perform :class:`~iris._data_manager.DataManager` instance equality.
+        Note that, this is explicitly not a lazy operation and will load any
+        lazy payload to determine the equality result.
+
+        Comparison is strict with regards to lazy or real managed payload,
+        the realised_dtype, the dtype of the payload and the payload content.
+
+        Args:
+
+        * other:
+            The :class:`~iris._data_manager.DataManager` instance to
+            compare with.
+
+        Returns:
+            Boolean.
+
+        """
+        result = NotImplemented
+
+        if isinstance(other, type(self)):
+            result = False
+            is_lazy = self.has_lazy_data() == other.has_lazy_data()
```

Comment:
I'll do a name change ... I've almost ran out of creative juice, so thanks for the suggestions :wink: